### ISSUE TITLE
Some reflections on Restangular and extensibility

### ISSUE THREAD

**Issue by ccollie** [mgonto/restangular#158]
TL;DR - because its based around a provider (rather than a factory), its difficult to develop Restangular based services in a "develop by feature" manner. 

First off, fantastic work. Even with the issues i've encountered, i'm likely to use it in an upcoming at-work project. 

I spent some time this weekend trying to create a pull request for issue

https://github.com/mgonto/restangular/issues/108

And ran into a wall. The issues stem from the fact that Restangular is provider based, which restricts the flexibility of what can happen after app init. In the specific case of issue 108, you cannot inject any service or factory into a provider, so $injector is not available. 
##### Background

In my controllers prefer semantic service methods like

```
Post.getActivities("users/jswift/inbox", {count:200, since:90389})
```

As opposed to

```
Restangular.one("users","jswift").getList('inbox', {count:200, since:90389})
```

So i encapsulate Restangular calls behind services. Some of these require configuration, for example per-resource transformers. Herein lies the issue - AFAIK all restangular configuration is forced to be during app init. And since i'm organizing by feature, i don't want to edit a global config every time i add a resource based service. Of course, i could add a module.config in the file defining the service, but it seems like boilerplate, and it comes with limitations on what services are available.

What i've had to do is to create a factory to deal with these issues, including personally keeping track of callbacks in closure global scope which makes them available to the global restangular configuration.
##### Another pain point

Sometime i need to restangularize parts of the server response.

For example, i have urls like

```
api/v1/users/jswift/feed
```

The result looks something like :

```
{
    items:[{id:700, actor: {id:100, display_name:'mary_mac'} }, ...],
    totalCount : 120,
    links : {
        next : {
            href : "users/potus/feed?since=1001"
        },
        prev : {
            href : "users/potus/feed?before=501"
        }
    },
}
```

 Good so far, except i need to restangularize "items". However there is no way to do that external to config. So i tried some nasty trickery to access the restangularizeElement method on the config. No luck.

Again, i'm not writing this to complain. This is moreso to outline some issues i
found in development so that we can find ways to ease friction using the library.
